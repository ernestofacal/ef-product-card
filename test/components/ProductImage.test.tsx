import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';
describe('ProductImage', () => {
  test('1-ProductImage debe de mostrar el componente con el nombre del producto 1', () => {
    const wrapper = renderer.create(<ProductImage img="https://ola.jpg" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  test('2-ProductImage debe de mostrar el componente correctamente con el titulo personalizado 1', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
