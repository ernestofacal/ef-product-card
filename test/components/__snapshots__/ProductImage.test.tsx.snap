// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductImage 1-ProductImage debe de mostrar el componente con el nombre del producto 1 1`] = `
<img
  alt="Product"
  className="productImg undefined"
  src="https://ola.jpg"
/>
`;

exports[`ProductImage 2-ProductImage debe de mostrar el componente correctamente con el titulo personalizado 1 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
